### ISSUE TITLE
Unused code elimination and deduplication is not applied when using sass's api: 'legacy' setting

### ISSUE THREAD

**Issue by AndrewJDR** [webpack/webpack#18333]
I just wanted to document this for others in the future, because I wasn't able to find this particular solution when I searched. 
TLDR: The solution is [to use the `api: "modern"` setting](https://webpack.js.org/loaders/sass-loader/#api) when using sass-loader with dart-sass if you want deduplication/dead code elimination on your sass/css.
Because this involves an interaction between webpack and sass, I also opened an issue on the dart-sass project - https://github.com/sass/dart-sass/issues/2221

Here is further information in case you are interested:

I use webpack 5 on a project, and recently switched from node-sass/libsass to dart-sass. The project imports fontawesome sass files in a .ts files in this manner:
```
import "@fortawesome/fontawesome-pro/scss/fontawesome.scss"
import "@fortawesome/fontawesome-pro/scss/brands.scss"
import "@fortawesome/fontawesome-pro/scss/solid.scss"
import "@fortawesome/fontawesome-pro/scss/regular.scss"
import "@fortawesome/fontawesome-pro/scss/light.scss"
import "@fortawesome/fontawesome-pro/scss/v4-shims.scss"
import "@fortawesome/fontawesome-pro/scss/thin.scss"
```

After the changeover from node-sass to dart-sass, the parsed webpack bundle size was much larger than before. It went from 5.64MB to 7.74MB.

**After some digging around, I found that specifying the modern API in the webpack configuration for the sass-loader reduced the bundle size back down to 5.64 MB.**
```
            {
              loader: "sass-loader",
              options: {
                api: "modern",
              }
            }
```
If the `api` setting is left unspecified, it defaults to "legacy", and fontawesome and other sass imports don't seem to receive any "dead-code elimination" or deduplication, leading to larger bundle sizes. Specifically, I observed with webpack-bundle-analyzer that each imported sass file like "brands.scss" and "solid.scss" consume nearly 400KB in the bundle because they're duplicating many of the same sass file imports that were already imported by prior file imports like fontawesome.scss. Setting api to "modern" seems to ensure that some form of "dead code elimination" and deduplication is applied, and files like "brands.scss" and "solid.scss" only contribute a few KB to the bundle size, and even fontawesome.scss (the first import) contributes around 250KB rather than about 600KB.

Snippets of the webpack bundle analyzer for the fontawesome sass files are shown below.

Using `api: "modern"`:
![image](https://github.com/sass/dart-sass/assets/1588857/3c93e1fa-693b-4eb4-9fd5-f68d82fe15e4)

Using `api: "legacy"` (the default):
![image](https://github.com/sass/dart-sass/assets/1588857/5aab507d-6756-4829-9d5d-9a2efd8feeba)
